src/components/AdvancedSearchBar/AdvancedSearchBar.vue

```diff
@@ -0,0 +1,181 @@
+            {{ totalResults }} {{ t('search.matchingResults') }}
+          </div>
+          <div class="float-right flex">
+            <LobLink
+              :to="link"
+              :underline="false"
+              class="text-sm hover:text-primary-900"
+              @click="hide"
+            >
+              {{ t('search.seeAllResults') }}
+              <Icon class="inline" :icon="IconName.NEXT" />
+            </LobLink>
+          </div>
+        </div>
+      </template>
+      <!-- If no results are found show a no results message -->
+      <template v-else>
+        {{ t('search.noResults') }}
+      </template>
+    </div>
+  </div>
+</template>
+<script setup lang="ts">
+import TextInput from '../TextInput/TextInput';
+import LobTable from '../Table/Table';
+import TableHeader from '../Table/TableHeader';
+import TableBody from '../Table/TableBody';
+import TableRow from '../Table/TableRow';
+import LobLink from '../Link/Link.vue';
+import MagnifyingGlass from '../Icons/MagnifyingGlass';
+import XmarkLarge from '../Icons/XmarkLarge';
+import { ref, watch, computed, onMounted, onUnmounted } from 'vue';
+import { Icon, IconName } from '../Icon';
+
+const searchTerm = ref('');
+const searchResults = ref([]);
+const searching = ref(false);
+const visible = ref(false);
+const timeout = ref(null);
+const searchBar = ref(null);
+
+const props = withDefaults(
+  defineProps<{
+    searchFunction: Function;
+    count?: number;
+    link?: string;
+    footer?: boolean;
+  }>(),
+  {
+    count: 0,
+    link: '',
+    footer: true
+  }
+);
+
+const disabled = computed(() => !searchTerm.value);
+const totalResults = computed(() => props.count);
+
+function search(searchTerm: string) {
+  searchResults.value = ref([]);
+  props
+    .searchFunction(searchTerm)
+    .then((results) => {
+      searchResults.value = results;
+    })
+    .finally(() => {
+      searching.value = false;
+    });
+}
+
+function debounceSearch(searchTerm: string, delayMs: number = 500) {
+  searching.value = true;
+  clearTimeout(timeout);
+  timeout.value = setTimeout(async () => {
+    search(searchTerm);
+  }, delayMs);
+}
+
+watch(searchTerm, async (newSearchTerm) => {
+  if (newSearchTerm && newSearchTerm !== '') {
+    visible.value = true;
+    debounceSearch(newSearchTerm);
+  }
+});
+
+function clearSearch() {
+  if (searchTerm.value) {
+    searchTerm.value = '';
+    searchResults.value = [];
+    visible.value = false;
+  }
+}
+
+function hide() {
+  visible.value = false;
+}
+
+function onClickOutside($event) {
+  if (searchBar.value) {
+    const clickOnTheContainer = searchBar.value === $event.target;
+    const clickOnChild =
+      searchBar.value && searchBar.value.contains($event.target);
+
+    if (!clickOnTheContainer && !clickOnChild) {
+      hide();
+    }
+  }
+}
+
+onMounted(() => {
+  window.addEventListener('click', onClickOutside);
+});
+
+onUnmounted(() => {
+  window.removeEventListener('click', onClickOutside);
+});
```

Comment:
This doesn't seem like the Vue way of doing things as its using plain JS event handlers. This could justify a new Vue hook with reactive state attached to it. This also seems like there is a concern for accessibility since people who don't utilize a mouse can't click outside (I would think focus would control this)

Feel free to ignore this one as it may be a decent lift to do some of this but wanted note it